tests/library-static-only/BUILD

```diff
@@ -0,0 +1,17 @@
+package(default_testonly = 1)
+
+load(
+    "@io_tweag_rules_haskell//haskell:haskell.bzl",
+    "haskell_library",
+)
+
+haskell_library(
+    name = "library-static-only",
+    srcs = ["Lib.hs"],
+    visibility = ["//visibility:public"],
+    deps = [
+        "@hackage//:base",
+        "@hackage//:bytestring",
+    ],
+    linkstatic = True,
+)
```

Comment:
We could also add a `sh_test` that checks whether the resulting library is actually linked (semi-)statically.

Reply:
We could. But I ran out of brain power when trying to figure out how to do it very simply. It *might* be worth it if you know how to verify `ldd` output from Bazel without too much fuss.